File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
      const idx = {};
  for (let i = 0; i < numbers.length; i++) {
    const idx2 = target - numbers[i];
    if (idx2 in idx) {
      return [idx[idx2] + 1, i + 1];
    }
    idx[numbers[i]] = i;
  }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
         int left = 0;
    int right = numbers.length - 1;
    while (numbers[left] + numbers[right] != target) {
      if (numbers[left] + numbers[right] > target) {
        right--;
      }
      if (numbers[left] + numbers[right] < target) {
        left++;
      }
    }
    int[] output = { left + 1, right + 1 };
    return output;
    }
}